packages/data/src/components/use-select/test/index.js

```diff
@@ -1025,4 +1025,73 @@ describe( 'useSelect', () => {
 			expect( TestComponent ).toHaveBeenCalledTimes( 2 );
 		} );
 	} );
+
+	describe( 'usage without dependencies array', () => {
+		function registerStore( name, initial ) {
+			registry.registerStore( name, {
+				reducer: ( s = initial, a ) => ( a.type === 'inc' ? s + 1 : s ),
+				actions: { inc: () => ( { type: 'inc' } ) },
+				selectors: { get: ( s ) => s },
+			} );
+		}
+
+		it( 'does not memoize the callback when there are no deps', () => {
+			registerStore( 'store', 1 );
+
+			const Status = ( { multiple } ) => {
+				const count = useSelect(
+					( select ) => select( 'store' ).get() * multiple
+				);
+				return <div role="status">{ count }</div>;
+			};
+
+			const App = ( { multiple } ) => (
+				<RegistryProvider value={ registry }>
+					<Status multiple={ multiple } />
+				</RegistryProvider>
+			);
+
+			const rendered = render( <App multiple={ 1 } /> );
+			expect( rendered.getByRole( 'status' ) ).toHaveTextContent( 1 );
+
+			rendered.rerender( <App multiple={ 2 } /> );
+			expect( rendered.getByRole( 'status' ) ).toHaveTextContent( 2 );
```

Comment:
Maybe we should add inline comments here. It's a little hard to see at a glance how this checks for not memoized callbacks.

My understanding is that if we memoized callbacks by default (missing deps means no deps - `[]`), the second assertion would have failed. Am I following the logic here correctly?

Reply:
Added a comment in 84d52c92b2. Yeah, if the default deps were `[]`, like:
```js
useSelect( ( select ) => select( 'store' ).get() * multiple, [] );
```
then there is a "stale closure problem": the old value of `multiple` is bound inside the callback and continues to be used even when the prop is updated. The test checks that this doesn't happen.

Reply:
That was my guess as well. Thank you, @jsnajdr.